jetty-core/jetty-http3/jetty-http3-qpack/src/main/java/org/eclipse/jetty/http3/qpack/QpackDecoder.java

```diff
@@ -184,6 +185,8 @@ public boolean decode(long streamId, ByteBuffer buffer, Handler handler) throws
             // Decode it straight away if we can, otherwise add it to the list of EncodedFieldSections.
             if (requiredInsertCount <= insertCount)
             {
+                // TODO: if this throws session exception we really need to send the cancellation instruction.
```

Comment:
What is the intention for this? to do this in this PR or to open an issue to fix it later?

Reply:
There is issue https://github.com/jetty/jetty.project/issues/7676 already open for this.

Ideally I would hope to get this fixed in this PR, or at least soon after. Because I think it really is necessary for the upper layer to call `_qpackDecoder.streamCancellation(streamId)`. I'll need to talk to @sbordet about this.